### ISSUE TITLE
completion prefix behavior regression

### ISSUE THREAD

**Issue by floam** [fish-shell/fish-shell#3483]
Something strange is happening with filename completions.

## Reproduction steps
1. Type `ls`
2. Press **tab**

I expect to just see filenames to complete. But every item in the pager has the command (ls) as a bolded prefix. This seems to have changed recently. (?)

<img width="452" alt="screenshot 2016-10-23 at 5 35 30 pm" src="https://cloud.githubusercontent.com/assets/291142/19630871/ba31c652-9947-11e6-8ebb-79dcdb307c27.png">

**Comment by floam** [comment]
A few commits back does this. I'll bisect.
<img width="458" alt="screenshot 2016-10-23 at 5 40 25 pm" src="https://cloud.githubusercontent.com/assets/291142/19630879/db80d0fa-9947-11e6-9522-860606cd1437.png">

**Comment by floam** [comment]
Bisected to 42458ff7ab358475488db7c51fbff121f522d78a

**Comment by krader1961** [comment]
This is due to my change to the `parse_util_get_parameter_info()` function to eliminate pointless nested `if ()` conditionals. I made an incorrect logical transform when attempting to simplify that code; specially a `==` test that should have been converted to `!=`. Having said that I am extremely annoyed by the original code and my oclint inspired change to eliminate useless nested conditionals. The logic of this code is extremely opaque with or without my change. So I'm going to ponder this for another day before merging a change to fix the bug I introduced.
